frontstage/controllers/conversation_controller.py

```diff
@@ -34,11 +34,14 @@ def get_conversation(thread_id):
         response = session.get(url, headers=headers)
         try:
             response.raise_for_status()
-        except HTTPError:
-            raise ApiError(logger, response,
-                           log_level='error',
-                           message='Thread retrieval failed',
-                           thread_id=thread_id)
+        except HTTPError as exception:
+            if exception.response.status_code == 404:
```

Comment:
I think this should be 403

Reply:
It _should be_, but I think Chris right that it's actually not.

The service, IIRC, throws back 'Not found' so that fuzzing the server to find real values won't produce different results, but I may be wrong.

@ChristopherMShelton to confirm?

Reply:
I'm with Matt, this should be a 403.  If the service returns 404 if there is a message/thread but they don't have access, then it should be fixed in secure-message, worked around here.

Reply:
I think the bit of code that needs to change can be found https://github.com/ONSdigital/ras-secure-message/blob/master/secure_message/repository/retriever.py#L190

At the minute, the thread_id AND the user_id need to match, otherwise it returns nothing in the form of a 404 which isn't true at all.  The API should give back an honest answer, and frontstage should then make the choice if they want to hide the existence of an unauthorised bit of data with a 404 (which we're choosing not to do with this PR!).

The code for this will look something like:
```python
result = SecureMessage.query.join(Conversation) \
                .join(Status) \
                .filter(SecureMessage.thread_id == thread_id) 

# First check if a thread exists           
if not result.all():
    logger.error('Thread does not exist', thread_id=thread_id, user_id=user.user_uuid)
    raise NotFound(description=f"Conversation with thread_id '{thread_id}' does not exist")

result = result.filter(Status.actor == user.user_uuid) \
                .order_by(SecureMessage.id.desc())

# Then we check if the user matches.  If the filter returns nothing, then it's for someone else.
if not result.all():
    logger.error('Thread but exists, but belongs to a different respondent', thread_id=thread_id, user_id=user.user_uuid)
    raise NotAuthorised(description=f"User {user_id} not authorised to view conversation with thread_id '{thread_id}'")
```

Don't copy that verbatim as the logging is awful, but it's to show what I mean.  You'd need to define a 'notAuthorised' exception if it doesn't exist already and have it return a 403 when it's raised.